install/autodetect.go

```diff
@@ -57,7 +57,40 @@ func (k *K8sUninstaller) autodetect(ctx context.Context) error {
 	return nil
 }
 
-func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
+func (k *K8sInstaller) detectDatapathMode(withKPR bool) {
+	if k.params.DatapathMode == "" {
```

Comment:
Instead of this if/else, can you just early return if DataPath  mode is set? Should type out a bit cleaner.